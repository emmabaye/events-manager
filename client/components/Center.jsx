import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import DeleteModal from './DeleteModal.jsx'; // eslint-disable-line


/**
 * React component for center card
 * displayed in admin panel
 */
export default class Center extends Component {
  /**
   * React's method to render react component.
   *
   * @return {object}
   */
  render() {
    return (
      <div className="col-md-4 event" style={{ height: "300px" }}>
        <div className="card">
          <div className="card-block" style={{ height: "300px" }}>
            <h5 className="card-title"><b>{this.props.centerDetails.name}</b></h5>
            <p className="card-text h6" >{`${this.props.centerDetails.description.substr(0, 50)}...`}</p>
            <p className="card-text h6"><b>Location</b>:{`${this.props.centerDetails.location.substr(0, 25)}...`}</p>
            <p className="card-text h6"><b>Capacity</b>: {this.props.centerDetails.capacity} seats</p>
            <p className="card-text h6"><b>Price</b>: &#8358;{this.props.centerDetails.price}</p>
            <p className="card-text">
              <b>Availability</b>: {
                (this.props.centerDetails.available === 'true') ? "Center is Available" : "Not Available"
              }
            </p>
            <p className="text-center">
              <Link to="#" className="btn btn-sm btn-primary"
                onClick={() => this.props.show('centerDetails', this.props.centerDetails.id)} >
                <i className="fa fa-info-circle fa-lg" /> View Center
              </Link>
              <Link to="#" className="btn btn-sm btn-success"
                onClick={() => this.props.show('modifyCenter', this.props.centerDetails.id)} >
                <i className="fa fa-edit fa-lg" /> Modify
              </Link>
              <Link to="#" className="btn btn-sm btn-danger" data-toggle="modal"
                data-target={`#${this.props.centerDetails.id}`}><i className="fa fa-trash-o fa-lg" /> Delete
              </Link>
            </p>
          </div>
        </div>
        <DeleteModal item="center" itemId={this.props.centerDetails.id} show={this.props.show} />
      </div>
    );
  }
}
