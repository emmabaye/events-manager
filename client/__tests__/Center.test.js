import { BrowserRouter } from 'react-router-dom';
import Center from '../components/Center.jsx';

describe('Center Component', () => {
  const props = {
    centerDetails: {
      id: "1",
      name: "",
      description: {
        substr: () => {}
      },
      location: {
        substr: () => {}
      },
      capacity: "",
      price: "",
      available: 'true'
    },
    show: () => {}
  };
  const initialState = {
    eventReducer: {
      userEvents: []
    },
    centerReducer: {
      allCenters: []
    }
  };
  const store = mockStore(initialState);

  it('it should render Center component', () => {
    const wrapper = mount(<BrowserRouter><Provider store={store} ><Center {...props} /></Provider></BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render Center component', () => {
    props.centerDetails.available = 'false';
    const wrapper = mount(<BrowserRouter><Provider store={store}><Center {...props} /></Provider></BrowserRouter>);
    expect(wrapper.length).toEqual(1);
  });

  /*
 it('it should handle onClick event', () => {
    const wrapper = shallow(<BrowserRouter><Provider store={store} ><Center {...props} /></Provider></BrowserRouter>);
    let viewCenterButton = wrapper.find('.btn-primary');
    let submitButton = wrapper.find('.btn-success');
    viewCenterButton.simulate('click');
    submitButton.simulate('click');
  });
  */
});
