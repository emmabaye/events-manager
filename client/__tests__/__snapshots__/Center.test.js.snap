// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component it should render Center component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Center
        centerDetails={
          Object {
            "available": "true",
            "capacity": "",
            "description": Object {
              "substr": [Function],
            },
            "id": "1",
            "location": Object {
              "substr": [Function],
            },
            "name": "",
            "price": "",
          }
        }
        show={[Function]}
      >
        <div
          className="col-md-4 event"
          style={
            Object {
              "height": "300px",
            }
          }
        >
          <div
            className="card"
          >
            <div
              className="card-block"
              style={
                Object {
                  "height": "300px",
                }
              }
            >
              <h5
                className="card-title"
              >
                <b />
              </h5>
              <p
                className="card-text h6"
              >
                undefined...
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Location
                </b>
                :
                undefined...
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Capacity
                </b>
                : 
                 seats
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Price
                </b>
                : ₦
              </p>
              <p
                className="card-text"
              >
                <b>
                  Availability
                </b>
                : 
                Center is Available
              </p>
              <p
                className="text-center"
              >
                <Link
                  className="btn btn-sm btn-primary"
                  onClick={[Function]}
                  replace={false}
                  to="#"
                >
                  <a
                    className="btn btn-sm btn-primary"
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-info-circle fa-lg"
                    />
                     View Center
                  </a>
                </Link>
                <Link
                  className="btn btn-sm btn-success"
                  onClick={[Function]}
                  replace={false}
                  to="#"
                >
                  <a
                    className="btn btn-sm btn-success"
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-edit fa-lg"
                    />
                     Modify
                  </a>
                </Link>
                <Link
                  className="btn btn-sm btn-danger"
                  data-target="#1"
                  data-toggle="modal"
                  replace={false}
                  to="#"
                >
                  <a
                    className="btn btn-sm btn-danger"
                    data-target="#1"
                    data-toggle="modal"
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-trash-o fa-lg"
                    />
                     Delete
                  </a>
                </Link>
              </p>
            </div>
          </div>
          <Connect(DeleteModal)
            item="center"
            itemId="1"
            show={[Function]}
          >
            <DeleteModal
              allCenters={Array []}
              dispatch={[Function]}
              item="center"
              itemId="1"
              myEvents={Array []}
              show={[Function]}
            >
              <div
                aria-hidden="true"
                aria-labelledby="11Label"
                className="modal fade"
                id="1"
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h6
                        className="modal-title"
                        id="eventModalLabel"
                      >
                        Delete 
                        center
                      </h6>
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      <p
                        className="text-center"
                      >
                        Are you sure you want to delete this 
                        center
                        ?
                      </p>
                      <form />
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-sm btn-secondary"
                        data-dismiss="modal"
                        type="button"
                      >
                        Cancel
                      </button>
                      <button
                        className="btn btn-sm btn-danger"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete 
                        center
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </DeleteModal>
          </Connect(DeleteModal)>
        </div>
      </Center>
    </Provider>
  </Router>
</BrowserRouter>
`;
